specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-04-01-preview/managedHsm.json

```diff
@@ -1389,6 +1439,76 @@
         }
       },
       "description": "Properties of a private link resource."
+    },
+    "CheckMhsmNameAvailabilityResult": {
+      "properties": {
+        "nameAvailable": {
+          "readOnly": true,
+          "type": "boolean",
+          "description": "A boolean value that indicates whether the name is available for you to use. If true, the name is available. If false, the name has already been taken or is invalid and cannot be used."
+        },
+        "reason": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The reason that a managed hsm name could not be used. The reason element is only returned if NameAvailable is false.",
+          "enum": [
+            "AccountNameInvalid",
+            "AlreadyExists"
+          ],
+          "x-ms-enum": {
+            "name": "Reason",
+            "modelAsString": false
+          }
+        },
+        "message": {
+          "readOnly": true,
+          "type": "string",
+          "description": "An error message explaining the Reason value in more detail."
+        }
+      },
+      "description": "The CheckMhsmNameAvailability operation response."
+    },
+    "MHSMSecurityDomainProperties":{
```

Comment:
`ManagedHSMSecurityDomainProperties` might be better. `MHSM` would likely produce strange name in client.

Reply:
Addressed in latest commit